core/admin/mailu/sso/views/base.py

```diff
@@ -48,6 +49,81 @@ def login():
+
+    fields = []
+    if str(app.config["WEBMAIL"]).upper() != "NONE":
+        fields.append(form.submitWebmail)
+    if str(app.config["ADMIN"]).upper() != "FALSE":
+        fields.append(form.submitAdmin)
+    fields = [fields]
+
+    if str(target).upper()=="ADMIN":
+        destination = "0; url="+ app.config['WEB_ADMIN']
+    else:
+        destination = "0; url="+ app.config['WEB_WEBMAIL']
+
+    payload_dict = {
+        'Refresh': destination
+    }
+
+    device_cookie, device_cookie_username = utils.limiter.parse_device_cookie(flask.request.cookies.get('rate_limit'))
+    if username != device_cookie_username and utils.limiter.should_rate_limit_ip(client_ip):
+        flask.flash('Too many attempts from your IP (rate-limit)', 'error')
+        return flask.render_template('login.html', form=form, fields=fields)
+    if utils.limiter.should_rate_limit_user(username, client_ip, device_cookie, device_cookie_username):
+        flask.flash('Too many attempts for this user (rate-limit)', 'error')
+        return flask.render_template('login.html', form=form, fields=fields)
+
+    if not (app.config['PROXY_SECRET'] in (None, '') or not app.config['PROXY_SECRET']) and app.config['PROXY_SECRET'] == secret and username != "":
+        user = models.User.get(flask.request.headers.get('X-Auth-Email'))
+        if user:
+            flask.session.regenerate()
+            flask_login.login_user(user)
+            return flask.render_template('login.html', form=form, fields=fields), payload_dict
+        else:
+            if app.config['PROXY_CREATE']:
+                if domain.has_email(localpart):
+                    flask.flash('Email is already used', 'error')
+                    return flask.redirect(flask.url_for('.login'))
+                else:
+                    #Create a user
+                    user = models.User(
+                        localpart=localpart,
+                        domain=domain,
+                        global_admin=False
+                    )
+                    user.set_password(secrets.token_urlsafe(32))
+                    models.db.session.add(user)
+                    models.db.session.commit()
+                    user.send_welcome()
+                    flask.session.regenerate()
+                    flask_login.login_user(user)
+                    flask.current_app.logger.info(f'Login succeeded by proxy created user: {username} from {client_ip}.')
+                    return flask.render_template('login.html', form=form, fields=fields), payload_dict
+            else:
+                utils.limiter.rate_limit_user(username, client_ip, device_cookie, device_cookie_username) if models.User.get(username) else utils.limiter.rate_limit_ip(client_ip)
+                flask.current_app.logger.warn(f'Login failed by proxy for {username} from {client_ip}.')
+                flask.flash('Auth by proxy failed - try to login by credentials', 'error')
```

Comment:
This doesn't make sense to me: if the PROXY_SECRET token matches it's not a malicious attempt, merely a misconfiguration... and rate-limiting non-existing users just won't work.

Reply:
Removed